File: backend/app/schemas/analyze.py
"""分析API用スキーマ"""
from pydantic import BaseModel
from datetime import date
from typing import Optional


class WeeklyReport(BaseModel):
    """一週間レポート"""
    start_date: date
    end_date: date
    total: int
    count: int
    average: float


class AnalyzeResponse(BaseModel):
    user: str
    total: int
    budget: int | None
    remaining: int | None
    remaining_rate: float | None
    pace_rate: float | None
    coach_type: str
    coach_message: str
    has_expenses: bool
    has_budget: bool
    weekly_report: WeeklyReport


class AIAnalyzeResponse(BaseModel):
    """AI分析APIレスポンス"""
    user: str
    total: int
    budget: int
    coach_type: str
    ai_message: str

File: frontend/lib/types/analyze.ts
export interface AnalyzeResponse {
  user: string;
  total: number;
  budget: number | null;
  remaining: number | null;
  remaining_rate: number | null;
  pace_rate: number | null;
  coach_type: string;
  coach_message: string;
  has_expenses: boolean;
  has_budget: boolean;
  weekly_report: WeeklyReport;
}

export interface WeeklyReport {
  start_date: string;
  end_date: string;
  total: number;
  count: number;
  average: number;
  coach_message: string;
}

export interface AIAnalyzeResponse {
  user: string;
  total: number;
  budget: number;
  coach_type: string;
  ai_message: string;
}
